tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
+    function_leapp_cv = function_leapp_cv.read()
+
+
+@pytest.fixture
+def custom_leapp_host(upgrade_path):
+    deploy_args = {}
+    deploy_args['deploy_rhel_version'] = upgrade_path['source_version']
+    logger.info('Creating Leapp Host - RHEL %s', deploy_args)
+    with Broker(
+        workflow='deploy-rhel',
+        host_class=ContentHost,
+        deploy_rhel_version=upgrade_path['source_version'],
+        deploy_flavor=settings.flavors.default,
+    ) as chost:
+        yield chost
+
+
+@pytest.mark.parametrize(
+    'upgrade_path',
+    [
+        # {'source_version': RHEL7_VER, 'target_version': RHEL8_VER},
+        {'source_version': RHEL8_VER, 'target_version': RHEL9_VER},
+    ],
+    ids=lambda upgrade_path: f'{upgrade_path["source_version"]}'
+    f'_to_{upgrade_path["target_version"]}',
+)
+@pytest.mark.usefixtures(
+    'leapp_sat_content',
+    'register_host_with_satellite',
+    'verify_target_repo_on_satellite',
+    'precondition_check_upgrade_and_install_leapp_tool',
+    'fix_inhibitors',
+)
+def test_leapp_upgrade_rhel(
+    module_target_sat,
+    custom_leapp_host,
+    upgrade_path,
+):
+    """Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+
+    :id: 8eccc689-3bea-4182-84f3-c121e95d54c3
+
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create LCE, Create CV, add repositories to it, publish and promote CV, Create AK, etc.
+        3. Register content host with AK
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    logger.info('Running test for upgrade path %s', upgrade_path)
+    old_ver = custom_leapp_host.os_version.major
+    # 6. Run LEAPP-PREUPGRADE Job Template-
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run preupgrade via Leapp"'})[0]
+        .id
+    )
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {custom_leapp_host.hostname}',
+        },
+    )
+    module_target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1800
+    )
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    # In case of Upgrade Path A - Rehl-7 to Rhel-8
+    # Inhibitors should fix after running LEAPP-PREUPGRADE Job Template
```

Comment:
- Few inhibitors come after executing `LEAPP-PREUPGRADE` Job template
- In case of rhel-7.9, if we fix any inhibitor before executing `LEAPP-PREUPGRADE`, it disturbs the flow of execution and gives some more inhibitors after execution of `LEAPP-UPGRADE` Job template
- To maintain flow, inhibitors need to be fix as per this order only
- It doesn't looks good to add that stuff related to rhel-7 inside test module